### ISSUE TITLE
Invalid IDE0059 (value never used) with using and local function capture

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#33843]
**Version Used**: 
16.0.0 RC

**Steps to Reproduce**:

```csharp
private class Disposable: IDisposable { public void Dispose() { } }
public int M()
{
    var result = 0;
    void append() => result += 1; // IDE0059 for 'result'
    using (var a = new Disposable())
        append();
    return result;
}
```

**Expected Behavior**:
No message, everything is used as expected.

**Actual Behavior**:
IDE0059 is triggered within the local function. Applying the code fix changes the local function to the invalid
```csharp
void append() => 1;
```

Not only the using is required, but also the assignment within. Removing `var a = ` from the sample also removes IDE0059.

**Comment by jinujoseph** [comment]
@mavasani can you dupe this pls

**Comment by mavasani** [comment]
This still repros on Preview4. Interesting that this only repros in presence of the using statement. The generated CFG has a synthesized try/finally and likely something about the flow walker or analysis is not detecting that the write from `result +=1` is reachable from `return result`. I will investigate.
